icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java

```diff
@@ -4129,11 +4128,46 @@ static void generateCXXHConstants(String thisVersion) {
                     System.out.println("#endif /* U_HIDE_DRAFT_API */");
                 }
                 System.out.println("");
+                // Hack: METRIC-TON unit changed its name from "metric-ton" to "tonne"
+                // In order to preserve the existing APIs for "metric-ton" we need to
+                // add those APIs manually
+                if (name.equals("Tonne")) {
+                    addCXXHForMetricTon();
+                }
             }
         }
         System.out.println("// End generated createXXX methods");
     }
 
+    // Add the headers for "metric-ton"
+    // The tool won't create them any more    
+    private static void addCXXHForMetricTon() {
+        System.out.println("    /**");
+        System.out.println("     * Returns by pointer, unit of mass: metric-ton");
+        System.out.println("     * (renamed to tonne in CLDR 42 / ICU 72).");
+        System.out.println("     * Caller owns returned value and must free it.");
+        System.out.println("     * Note: In ICU 74 this will be deprecated in favor of");
+        System.out.println("     * createTonne(), which is currently draft but will");
+        System.out.println("     * become stable in ICU 74, and which uses the preferred naming.");
```

Comment:
@DraganBesevic in a follow-up PR please fix these docs (here and the next function), as discussed.